diff --git a/mk/packaging.mk b/mk/packaging.mk
index 436e933227..24106bb711 100644
--- a/mk/packaging.mk
+++ b/mk/packaging.mk
@@ -45,26 +45,21 @@ DIST_SUPPORT = $(foreach pkg, $(DIST_SUPPORT_PACKAGES), $(SUPPORT_SRC_DIR)/$(pkg
 DEB_BUILD_DEPENDS := libboost-dev, curl, m4, debhelper
 DEB_BUILD_DEPENDS += , fakeroot, python, libncurses5-dev, libcurl4-openssl-dev
 
-ifneq ($(UBUNTU_RELEASE),)
-  ifneq ($(filter $(UBUNTU_RELEASE), trusty xenial),)
-    # RethinkDB fails to compile with GCC 6 (#5757)
-    DEB_BUILD_DEPENDS += , g++-5, libssl-dev
-    DSC_CONFIGURE_DEFAULT += CXX=g++-5
-  else
-    # RethinkDB fails to compile with GCC 6 (#5757) -- and there is
-    # no GCC 5 in later Ubuntus.  We need to use libssl1.0-dev to be
-    # compatible with libcurl when linking.
-    DEB_BUILD_DEPENDS += , clang, libssl1.0-dev
-    DSC_CONFIGURE_DEFAULT += CXX=clang++
-  endif
-else ifneq ($(DEB_RELEASE),)
-  ifneq ($(filter $(DEB_RELEASE), jessie),)
-    DEB_BUILD_DEPENDS += , g++, libssl-dev
-  else
-    # As with Ubuntus.
-    DEB_BUILD_DEPENDS += , clang, libssl1.0-dev
-    DSC_CONFIGURE_DEFAULT += CXX=clang++
-  endif
+# Pick a working C++ compiler
+ifneq (,$(findstring $(UBUNTU_RELEASE)$(DEB_RELEASE), trusty xenial yakkety jessie))
+  DEB_BUILD_DEPENDS += , g++5
+  DSC_CONFIGURE_DEFAULT += CXX=g++-5
+else ifneq (,$(findstring $(UBUNTU_RELEASE)$(DEB_RELEASE), clang))
+  DEB_BUILD_DEPENDS += , clang
+else
+  DEB_BUILD_DEPENDS += , g++
+endif
+
+# Pick the appropriate version of libssl
+ifneq (,$(findstring $(UBUNTU_RELEASE)$(DEB_RELEASE), trusty xenial jessie yakkety))
+  DEB_BUILD_DEPENDS += , libssl-dev
+else
+  DEB_BUILD_DEPENDS += , libssl1.0-dev
 endif
 
 ifneq (1,$(BUILD_PORTABLE))
